thunder/tests/test_fa3_executor.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+import torch
+
+import thunder
+import thunder.core
+from thunder.executors.fa3ex import fa3_ex
+from thunder.tests.framework import requiresCUDA
+
+# flash attn 3
+try:
+    from flash_attn_interface import _flash_attn_forward, _flash_attn_backward, flash_attn_func
+
+    HAS_FA3 = True
+except:
+    HAS_FA3 = False
+
+
+@pytest.mark.parametrize("device,", ["cuda"])
```

Comment:
Would you also add a test that the fa3 executor is not used when the inputs are unsupported? From the checker function and @crcrpar 's comment, that could inputs where the `attn_mask` is not None, the `dropout` is nonzero, the tensors are on the `cpu` device, and the dtype is torch.bfloat16 or torch.float32

Reply:
Good idea, added